app/src/types/profile.js

```diff
@@ -38,16 +53,34 @@ export type UpsertProfile = t.TypeOf<typeof UpsertProfileModel>;
  *   User data extracted from SPID.
  * @returns {Profile}
  */
-export function toAppProfile(from: GetProfileOKResponse, user: User): Profile {
-  return {
-    name: user.name,
-    family_name: user.family_name,
-    fiscal_code: user.fiscal_code,
-    email: from.email,
-    preferred_languages: from.preferredLanguages,
-    is_inbox_enabled: from.isInboxEnabled,
-    version: from.version
-  };
+export function toAppProfile(
+  from: GetProfileOKResponse | null,
+  user: User
+): ProfileWithEmail | ProfileWithoutEmail {
+  if (from === null) {
```

Comment:
accepting an optional value is a code smell, meaning the check should happen at the caller of this function (thus the else needs to happen at the caller)